### PR TITLE
Feature/139 external user

### PR DESCRIPTION
**Pull request by theblockstalk**
Main changes:
- issue #139 I have move 1 out of 3 functions from UserApps class to ExternalUser class
- changed enum KeyManagelLevel
- built integration test for full success SSO flow

### CODE REVIEW

**Comment by theblockstalk** [comment]
around commit [041d51f](https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/161/commits/041d51f17ce128bf7e33a8a7e3e54da8b8a88320) I have updated the Integration tests to also use JSDOM. This was done successfully in commit [b87ac65](https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/161/commits/b87ac6581f27964e1cefe72e5218e971cd2fe1f6).

However, it is only able to run with tsdx, not with pure jest. I have been trying to get it to run with pure jest, this is so that we can run it with VS Code tools such as debugging and the jest extensions which would be really powerful.

This seems to encounter several issues:
1. Because we are using a custom test environment to inject JSDOM with, when we use pure jest, we get an error saying that the custom test environment is missing getVmContext() function
https://stackoverflow.com/questions/70910410/testing-expo-with-enzyme-jest-environment-enzyme-does-not-export-getvmcontext/70943283#70943283
2. After trying to upgrade JSDOM as suggested in the above article we run into other issues
3. I have then tried to recreate the tsdx's version of jest.config.js, with automatic https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/161/files#diff-4007f7b8a1c2025a38e8e3b80aa05297f37ae3719c0f37255aa3c81febab50d3 and manual approaches https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/161/files#diff-0a9d68240f8e10eb15b5db41e8d5321a91ac6e628e6ab8eb1ae552e5cb433eba neither working fully
4. It seemed with the above approaches, that the transformers are not working with typescript properly
5. some background here  https://bobbyhadz.com/blog/typescript-jest-cannot-use-import-statement-outside-module

Next step ideas, because I'm moving on from trying this:
* try installing jest manually?
https://kulshekhar.github.io/ts-jest/docs/getting-started/installation/
* https://github.com/jaredpalmer/tsdx/issues/577
* https://kulshekhar.github.io/ts-jest/docs/getting-started/presets/
* do something in babel https://stackoverflow.com/questions/61781271/jest-wont-transform-the-module-syntaxerror-cannot-use-import-statement-outsi

**Comment by theblockstalk** [comment]
the plan is to merge this with the integration test as is, and then continuing to work on issue #139 in subsequent PRs. This is so others get the benefit of having this integration test

File: src/externalUser.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        };
+
+        // TODO use expiresIn to make JWT expire after 5 minutes
+
+        const signer = ES256KSigner(privateKey.data.array, true);
+        const jwk = await createJWK(publicKey);
+
+        const issuer = toDid(jwk);
+
+        const token = (await Message.sign(payload, { did: issuer, signer: signer as any, alg: 'ES256K-R' })).jwt;
```
Comment (Rebal67) [inline src/externalUser.ts:38]: if we want to move the external logic to here shouldn't we also move the sign Message and verify ?

File: src/services/errors.ts [hunk 2]
```diff
@@ -82,6 +86,8 @@ enum SdkErrors {
     InvalidKey = 'InvalidKey',
     invalidDataType = 'invalidDataType',
     missingChallenge = 'missingChallenge',
+    CommunicationNotConnected = 'CommunicationNotConnected',
+    CommunicationTimeout = 'CommunicationTimeout',
```
Comment (Rebal67) [inline src/services/errors.ts:90]: this needs to change in different repos as well

File: src/externalUser.ts (hunk 1 above)
Comment (theblockstalk) [reply src/externalUser.ts:38]: yes maybe, but in a future PR

**Pull request merged**